iodata/formats/molekel.py

```diff
@@ -223,13 +243,150 @@ def load_one(lit: LineIterator) -> dict:
-            None)
+            irrepsa + irrepsb)
 
     result = {
         'atcoords': atcoords,
         'atnums': atnums,
         'obasis': obasis,
         'mo': mo,
+        'atcharges': atcharges,
     }
     _fix_molden_from_buggy_codes(result, lit)
     return result
+
+
+@document_dump_one("Molekel", ['atcoords', 'atnums', 'mo', 'obasis'], ['atcharges'])
+def dump_one(f: TextIO, data: IOData):
+    """Do not edit this docstring. It will be overwritten."""
+    # Header
+    f.write('$MKL\n')
+    f.write('#\n')
+    f.write('# MKL format file produced by ORCA\n')
+    f.write('#\n')
+
+    # CHAR_MUL
+    f.write('$CHAR_MULT\n')
+    f.write('  {:.0f} {:.0f}\n'.format(data.charge, data.spinpol + 1))
+    f.write('$END\n')
+    f.write('\n')
+
+    # COORD
+    atcoords = data.atcoords / angstrom
+    f.write('$COORD\n')
+    for x in range(len(data.atnums)):
+        f.write('   {:d}   {: ,.6f}  {: ,.6f}  {: ,.6f}\n'.format(data.atnums[x], atcoords[x][0],
+                                                                  atcoords[x][1], atcoords[x][2]))
+    f.write('$END\n')
+    f.write('\n')
+
+    # CHARGES
+    if 'mulliken' in data.atcharges:
+        charges = data.atcharges['mulliken']
+    else:
+        charges = {}
+        warnings.warn('Mulliken charges not stored in IOData. Proceeding without printing them.')
+    f.write('$CHARGES\n')
+    for charge in charges:
+        f.write('  {: ,.6f}\n'.format(charge))
+    f.write('$END\n')
+    f.write('\n')
+
+    # BASIS
+    f.write('$BASIS\n')
+    iatom_last = 0
+    for shell in data.obasis.shells:
+        iatom_new = shell.icenter
+        if iatom_new != iatom_last:
+            f.write('$$\n')
+        for iangmom, (angmom, kind) in enumerate(zip(shell.angmoms, shell.kinds)):
+            iatom_last = shell.icenter
+            nbasis = len(CONVENTIONS[(angmom, kind)])
+            f.write(' {} {:1s} 1.00\n'.format(nbasis, angmom_its(angmom).capitalize()))
+            for exponent, coeff in zip(shell.exponents, shell.coeffs[:, iangmom]):
+                f.write('{:20.10f} {:17.10f}\n'.format(exponent, coeff))
+    f.write('\n')
+    f.write('$END\n')
+    f.write('\n')
+
+    if data.mo.kind == 'restricted':
+        # COEFF_ALPHA
+        f.write('$COEFF_ALPHA\n')
+        _dump_helper_coeffs(f, data, spin='a')
+
+        # OCC_ALPHA
+        f.write('$OCC_ALPHA\n')
+        _dump_helper_occ(f, data, spin='ab')
+
+    # Not taking into account generalized.
+    elif data.mo.kind == 'unrestricted':
+        # COEFF_ALPHA
+        f.write('$COEFF_ALPHA\n')
+        _dump_helper_coeffs(f, data, spin='a')
+
+        # OCC_ALPHA
+        f.write('$OCC_ALPHA\n')
+        _dump_helper_occ(f, data, spin='a')
+        f.write('\n')
+
+        # COEFF_BETA
+        f.write('$COEFF_BETA\n')
+        _dump_helper_coeffs(f, data, spin='b')
+
+        # OCC_BETA
+        f.write('$OCC_BETA\n')
+        _dump_helper_occ(f, data, spin='b')
+
```

Comment:
If you want to deal with the generalized case, you might add something like

```python
elif data.mo.kind == 'generalized':
    raise ValueError("The MKL format does not support generalized orbitals..")
```

Reply:
Thanks for your comment. Fixed it with an `else` statement to be more general.